src/main/scala/org/broadinstitute/hail/sparkextras/OrderedRDD.scala

```diff
@@ -422,6 +422,61 @@ class OrderedRDD[PK, K, V] private(rdd: RDD[(K, V)], val orderedPartitioner: Ord
       new OrderedRDD[PK, K, V](new BlockedRDD(rdd, newPartEnd), partitioner)
     }
   }
+
+  def filterIntervals(intervals: IntervalTree[PK]): OrderedRDD[PK, K, V] = {
+
+    import kOk.pkOrd
+    import kOk._
+    import Ordering.Implicits._
+
+    if (partitions.length <= 1) // FIXME: hacky
+      return this
+
+    if (intervals.isEmpty)
+      return OrderedRDD.empty[PK, K, V](rdd.sparkContext)
+
+
+    val intervalArray = intervals.toArray
+
+    val partitionIndexBuilder = mutable.ArrayBuilder.make[Int]
+
+    val rangeBounds = orderedPartitioner.rangeBounds
+
+    val nPartitions = partitions.length
+    var i = 0
+    while (i < nPartitions) {
+      if (i == 0) {
+        if (intervalArray.exists(_.start <= rangeBounds(0)))
+          partitionIndexBuilder += 0
+      } else if (i == nPartitions - 1) {
+        if (intervalArray.reverse.exists(_.end > rangeBounds.last))
+          partitionIndexBuilder += i
+      } else {
+        val lastMax = rangeBounds(i - 1)
+        val thisMax = rangeBounds(i)
+        // FIXME: loads a partition if the lastMax == interval.start.  Can therefore load unnecessary partitions
+        if (intervals.overlaps(Interval(lastMax, thisMax)) || intervals.contains(thisMax))
```

Comment:
basically, the range bound intervals you can get are left closed, right open.  Which is the exact opposite of the interval implementation we have.  It's very hard to do the right thing here and not be too conservative, because there's no way to "add epsilon" to any arbitrary ordered PK